tests/population/test_mortality.py

```diff
@@ -33,6 +51,169 @@ def test_mortality_default_lookup_configuration(
+    lookup_tables = mortality.lookup_tables
+    acmr = lookup_tables["all_cause_mortality_rate"](pop1.index)
+    modeled_csmr = mortality.cause_specific_mortality_rate(pop1.index)
+    unmodeled_csmr_raw = lookup_tables["unmodeled_cause_specific_mortality_rate"](pop1.index)
+    unmodeled_csmr = mortality.unmodeled_csmr(pop1.index)
+    expected_mortality_rates = (acmr - modeled_csmr - unmodeled_csmr_raw + unmodeled_csmr) * (
+        sim._clock.step_size.days / 365
+    )
+    # Cannot compare two floats with FuzzyChecker
+    assert np.isclose(mortality_rates, expected_mortality_rates).all()
+
+
+def test_mortality_updates_population_columns(setup_sim_with_pop_and_mortality):
+    sim, bp, mortality = setup_sim_with_pop_and_mortality
+    pop0 = sim.get_population()
+    sim.step()
+    pop1 = sim.get_population()
+
+    # Configrm mortalit7y component updated columns correctly
+    # Note alive will be tested by finding the simulants that died
+    columns_to_update = ["cause_of_death", "exit_time", "years_of_life_lost"]
+    dead_idx = pop1.index[pop1["alive"] == "dead"]
+    for col in columns_to_update:
+        assert (pop1.loc[dead_idx, col] != pop0.loc[dead_idx, col]).all()
+    assert (pop1.loc[dead_idx, "cause_of_death"] == "other_causes").all()
+    assert pop1.loc[dead_idx, "exit_time"].unique()[0] == sim._clock._clock_time
+
+
+def test_mortality_cause_of_death(
+    fuzzy_checker: FuzzyChecker,
+    base_config,
+    full_simulants,
+    base_plugins,
+    generate_population_mock,
+):
+
+    start_population_size = len(full_simulants)
+    generate_population_mock.return_value = full_simulants.drop(columns=["tracked"])
+    bp = BasePopulation()
+    mortality = Mortality()
+    # Set up Disease model
+    year_start = base_config.time.start.year
+    year_end = base_config.time.end.year
+
+    healthy = BaseDiseaseState("healthy")
+    mort_get_data_funcs = {
+        "dwell_time": lambda _, __: pd.Timedelta(days=0),
+        "disability_weight": lambda _, __: 0.0,
+        "prevalence": lambda _, __: build_table(
+            0.5, year_start - 1, year_end, ["age", "year", "sex", "value"]
+        ),
+        "excess_mortality_rate": lambda _, __: build_table(0.7, year_start - 1, year_end),
+    }
+    mortality_state = DiseaseState("sick", get_data_functions=mort_get_data_funcs)
+    healthy.add_transition(Transition(healthy, mortality_state))
+
+    model = DiseaseModel("test", initial_state=healthy, states=[healthy, mortality_state])
+    sim = InteractiveContext(
+        components=[bp, mortality, model], plugin_configuration=base_plugins, setup=False
+    )
+    override_config = {
+        "population": {
+            "population_size": start_population_size,
+            "include_sex": "Male",
+        },
+        "mortality": {
+            "all_cause_mortality_rate": {
+                "value": 0.8,
+            }
+        },
+    }
+    sim.configuration.update(override_config)
+    sim.setup()
+    sim.step()
```

Comment:
I think the target rate should be the mortality rate before taking the first time-step.